### ISSUE TITLE
podman generate kube - Error: no matching entries in passwd file

### ISSUE THREAD

**Issue by leonidas-o** [containers/podman#8474]
/kind bug

**Description**

Running on CentOS 8 and using the defaults repo Podman Version 1.6.4 and Buildah Version 1.11.6.

I'm getting the error `ERRO[0000] no matching entries in passwd file` when trying to generate a kube output from a pod with 4 containers.

**Steps to reproduce the issue:**

1. Create a pod with 4 containers.

2. Execute `podman generate kube MY_POD_ID`

**Describe the results you received:**

podman --log-level=debug generate kube POD_ID:
```
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/gitlab/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/gitlab/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/989                 
DEBU[0000] Using static dir /home/gitlab/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/989/libpod/tmp       
DEBU[0000] Using volume path /home/gitlab/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] overlay: mount_program=/usr/bin/fuse-overlayfs 
DEBU[0000] backingFs=xfs, projectQuotaSupported=false, useNativeDiff=false, usingMetacopy=false 
DEBU[0000] Initializing event backend journald          
DEBU[0000] using runtime "/usr/bin/runc"                
INFO[0000] running as rootless                          
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /home/gitlab/.local/share/containers/storage/libpod/bolt_state.db 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/gitlab/.local/share/containers/storage 
DEBU[0000] Using run root /run/user/989                 
DEBU[0000] Using static dir /home/gitlab/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /run/user/989/libpod/tmp       
DEBU[0000] Using volume path /home/gitlab/.local/share/containers/storage/volumes 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] Initializing event backend journald          
DEBU[0000] using runtime "/usr/bin/runc"                
DEBU[0000] Looking in container for user: FOO:FOO
ERRO[0000] no matching entries in passwd file 
```

**Describe the results you expected:**

A regular output (yml) like described in https://github.com/containers/podman/blob/master/docs/source/markdown/podman-generate-kube.1.md.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.6.4
RemoteAPI Version:  1
Go Version:         go1.13.4
OS/Arch:            linux/amd64
```

**Output of `buildah version`:**

```
Version:         1.11.6
Go Version:      go1.13.4
Image Spec:      1.0.1-dev
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  
image Version:   5.0.0
Git Commit:      
Built:           Thu Jan  1 01:00:00 1970
OS/Arch:         linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.13.4
  podman version: 1.6.4
host:
  BuildahVersion: 1.12.0-dev
  CgroupVersion: v1
  Conmon:
    package: conmon-2.0.6-1.module_el8.2.0+305+5e198a41.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.6, commit: a2b11288060ebd7abd20e0b4eb1a834bbf0aec3e'
  Distribution:
    distribution: '"centos"'
    version: "8"
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 985
      size: 1
    - container_id: 1
      host_id: 231073
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 989
      size: 1
    - container_id: 1
      host_id: 231073
      size: 65536
  MemFree: 7250616320
  MemTotal: 16605270016
  OCIRuntime:
    name: runc
    package: runc-1.0.0-65.rc10.module_el8.2.0+305+5e198a41.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 0
  SwapTotal: 0
  arch: amd64
  cpus: 8
  eventlogger: journald
  hostname: MYHOSTNAME
  kernel: 4.18.0-193.28.1.el8_2.x86_64
  os: linux
  rootless: true
  slirp4netns:
    Executable: /usr/bin/slirp4netns
    Package: slirp4netns-0.4.2-3.git21fdece.module_el8.2.0+305+5e198a41.x86_64
    Version: |-
      slirp4netns version 0.4.2+dev
      commit: 21fdece2737dc24ffa3f01a341b8a6854f8b13b4
  uptime: 374h 12m 45.12s (Approximately 15.58 days)
registries:
  blocked: null
  insecure: null
  search:
  - docker.io
  - registry.access.redhat.com
  - registry.redhat.io
store:
  ConfigFile: /home/gitlab/.config/containers/storage.conf
  ContainerStore:
    number: 7
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-0.7.2-5.module_el8.2.0+305+5e198a41.x86_64
      Version: |-
        fuse-overlayfs: version 0.7.2
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
  GraphRoot: /home/gitlab/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 33
  RunRoot: /run/user/989
  VolumePath: /home/gitlab/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.6.4-10.module_el8.2.0+305+5e198a41.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

No: latest version tested, as the goal is to use the latest available podman version from CentOS 8's default repos.
Yes: checked the troubleshooting guide, also found other similar github issues, but already closed. One stated that the error is solved with podman 1.6.4. But as it seems, it's not.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

On a physical server.

**Comment by mheon** [comment]
Can you provide a reproducer? It seems like one of the containers is doing
something sufficiently different as to cause this.

I will also note that we are no longer actively backporting fixes to 1.6.4,
which is now only receiving critical bug fixes and security issues. CentOS
will continue to roll forward to the latest Podman versions (8.3 will have
Podman 2.0) so I strongly encourage you to test with a more recent Podman.

**Comment by leonidas-o** [comment]
@mheon thanks for the quick reply. Ohh okay, I tried to generate the yaml output without my own container (which leaves the infra container, a postgres latest container and a redis latest container inside the pod) and it worked.

So it seems it could be a miss-created Image/Container. It's a multi stage build. I guess, the issue is related to the created user under `Run Image`? Do I need some extra work here to make podman happy?
```Dockerfile
# ================================
# Build image
# ================================
FROM swift:5.3-centos8 as build

RUN dnf -q upgrade -y
WORKDIR /build
COPY /Config/secrets/.ssh /root/.ssh/
COPY ./Package.* ./
RUN swift package resolve
COPY . .
RUN swift build --enable-test-discovery -c release
WORKDIR /staging
RUN cp "$(swift build --package-path /build -c release --show-bin-path)/Run" ./

# ================================
# Run image
# ================================
FROM swift:5.3-centos8-slim
RUN dnf -q upgrade -y
RUN useradd --user-group --create-home --system --skel /dev/null --home /app FOO
WORKDIR /app
COPY --from=build --chown=FOO:FOO /staging /app
USER FOO:FOO
EXPOSE 8080
ENTRYPOINT ["./Run"]
CMD ["serve", "--env", "production", "--hostname", "0.0.0.0", "--port", "8080"]

```

**Comment by rhatdan** [comment]
@leonidas-o Could you try this on a more recent Podman, preferable podma 2.2.1?

**Comment by leonidas-o** [comment]
2.2.1 is not available in the default CentOS 8 repo, for the time of writing 2.0.5 is the latest. Nevertheless I got an approach which is working for my ci/cd pipeline. I was able to create and run the pod without my own container in it (only containing two other official containers), that was working without any issues, then separately build and `podman run --pod=test-pod ...` my container within this test pod. 
I'm sorry not being able to investigate that a bit more for now. I will close the issue, thanks for your help.
